### ISSUE TITLE
IG-003: bounded-rational behavior policies and explicit mixed-strategy beliefs

### ISSUE THREAD

**Issue by Tristan-Stoltz-ERC** [Luminous-Dynamics/symthaea#3075]
## Parent

Child of #3054. Depends on qualified IG-001/IG-002 formal semantics.

## Goal

Introduce behavior-model interfaces for institutional experiments without pretending one policy family is "human rationality" or "AI rationality", and without mislabeling a one-step softmax/logit choice as a quantal-response equilibrium.

## Critical terminology boundary

Keep these distinct:

1. **Logit choice / quantal response function** — a local stochastic action rule mapping expected action utilities to probabilities.
2. **Quantal Response Equilibrium (QRE)** — a fixed point where every player's quantal response is consistent with the mixed strategies generating those expected utilities.
3. **Level-k / cognitive hierarchy** — bounded strategic-depth models with explicit beliefs about lower-level opponents.

IG-003 should implement (1) and the shared mixed-belief machinery first. A QRE fixed-point solver may be added only if convergence/residual semantics are explicit and independently tested; it must never return a bare `Qre=true` claim from a softmax call.

### `ActionDistribution`

Validated probability vector:

- non-empty;
- finite;
- every probability >= 0;
- normalized to 1 under an explicit tolerance/policy;
- no silent repair of grossly invalid mass.

### `MixedStrategyProfile`

One validated `ActionDistribution` per player, dimensions exactly matching `NPlayerGame::strategy_counts()`.

Initial semantics assume an independent product distribution across players. Correlated strategies require a future explicit type, not accidental reuse.

### Expected action utility

For player `i` and candidate action `a_i`, compute expected payoff against the product distribution of opponents while holding `i` to `a_i`.

This is a formal expectation over the declared finite game and belief profile; it is not an empirical behavioral claim.

## Behavior-policy interface

Prefer a small dependency-free interface conceptually equivalent to:

```text
expected action values -> validated ActionDistribution
```

The interface should be deterministic for fixed numeric inputs/model parameters even when the returned object represents stochastic choice probabilities.

#### Best response

Return probability mass only over utility-maximizing actions. Tie semantics must be explicit; prefer equal mass across exact/tolerance-declared ties rather than hidden lowest-index bias.

#### Logit choice

For finite precision `lambda >= 0`:

`p(a) ∝ exp(lambda * u(a))`

Use a numerically stable shifted softmax. `lambda = 0` should produce the uniform distribution. Non-finite precision is invalid.

Name this `LogitChoice` / `LogitResponse`, **not** `QRE`.

## Optional second stage: QRE solver

If included in this issue/child PR, return a structured convergence report with at least:

- precision parameters;
- initial mixed profile;
- iteration cap;
- tolerance;
- iterations performed;
- final profile;
- fixed-point residual;
- termination reason (`Converged`, `IterationLimit`, numeric failure, etc.).

A non-converged run is not an equilibrium.

## Level-k / cognitive hierarchy boundary

Do not implement these as labels on arbitrary policies. They require explicit:

- level-0 policy;
- opponent-level belief/distribution;
- recursion/depth bound;
- tie/response semantics.

If this makes IG-003 too broad, split Level-k/CH into IG-003B rather than weakening the model.

## Required controls

1. uniform opponent beliefs produce hand-computed expected action values in a small 2x2 fixture;
2. mixed-belief dimensions that do not match the game fail closed;
3. `LogitChoice(lambda=0)` is uniform;
4. high precision concentrates mass on the unique best action without overflow;
5. adding a constant to every action utility leaves logit probabilities invariant;
6. best-response ties receive the declared tie behavior;
7. invalid/NaN probabilities and precision fail closed;
8. if QRE is implemented, matching-pennies has the expected symmetric fixed point and non-convergence is represented explicitly rather than promoted.

## Authority boundary

Behavior policies are experimental model families. A mechanism performing well under them establishes robustness only to those declared models/parameters. It does not establish human usability, real-world strategyproofness, or AI-agent behavior.
